common/tools/dev-tool/src/util/customization/customize.ts

```diff
@@ -273,11 +273,46 @@ function copyCustomImports(customFile: SourceFile, originalFile: SourceFile) {
 
     const allImports = new Set<string>(originalNamedImports);
     for (const customNamedImport of customImport.getNamedImports()) {
-      allImports.add(customNamedImport.getName());
+      allImports.add(customNamedImport.getText());
     }
 
     originalImport?.removeNamedImports();
     originalImport?.addNamedImports(Array.from(allImports));
   }
   originalFile.organizeImports();
 }
+
+function commonPrefix(a: string, b: string) {
+  let i = 0;
+  while (i < a.length && i < b.length && a[i] === b[i]) i++;
+  return a.slice(0, i);
+}
+
+function isSelfImport(module: string, file: SourceFile): boolean {
+  const { customDir, originalDir } = getCustomizationState();
+  let projectPath = file.getDirectory();
+  while (projectPath.getRelativePathTo(customDir).startsWith("..")) {
+    projectPath = projectPath.getParent() as Directory;
+  }
+  // e.g: ./sources/generated/src
+  let relativeOriginal = originalDir.replace(/\\/g, '/').replace(projectPath.getPath(), ".");
+  // e.g: ./sources/customizations
+  let relativeCustom = customDir.replace(/\\/g, '/').replace(projectPath.getPath(), ".");
+  // e.g: ./sources/
+  let prefix = commonPrefix(relativeOriginal, relativeCustom);
```

Comment:
please use const for the ones that don't need to be reassigned

Reply:
fixed